<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>GWASapi User Guide</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">GWASapi User Guide</h1>



<p>The <a href="https://github.com/rqtl/GWASapi">GWASapi</a> package provides access to the <a href="https://www.ebi.ac.uk/gwas">NHGRI-EBI catalog of GWAS summary statistics</a>. For details on the API, see <a href="https://www.ebi.ac.uk/gwas/summary-statistics/docs/">its documentation</a>, as well as <a href="http://thebird.nl/">Pjotr Prins</a>â€™s <a href="https://github.com/pjotrp/racket-summary-stats">documentation at github</a>.</p>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>You can install GWASapi from <a href="https://github.com/rqtl/GWASapi">GitHub</a>.</p>
<p>You first need to install the <a href="https://github.com/hadley/devtools">devtools</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</a></code></pre></div>
<p>Then use <code>devtools::install_github()</code> to install GWASapi.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">install_github</span>(<span class="st">&quot;rqtl/GWASapi&quot;</span>)</a></code></pre></div>
<p>Load the package with <code>library()</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">library</span>(GWASapi)</a></code></pre></div>
</div>
<div id="lists-of-things" class="section level2">
<h2>Lists of things</h2>
<p>The purpose of the GWASapi package is to provide access to summary statistics for human GWAS. First, you can get lists of studies and traits that are available.</p>
<p>To get lists of studies, use <code>list_studies()</code>. The default is to return just 20 studies. You can control that limit with the argument <code>size</code>. You can also use <code>start</code> to step through the full set.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">list_studies</span>(<span class="dt">size=</span><span class="dv">5</span>)</a></code></pre></div>
<pre><code>## [1] &quot;GCST000028&quot; &quot;GCST000392&quot; &quot;GCST000568&quot; &quot;GCST000569&quot; &quot;GCST000571&quot;</code></pre>
<p>To retrieve <em>all</em> studies, set a higher limit</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">all_studies &lt;-<span class="st"> </span><span class="kw">list_studies</span>(<span class="dt">size=</span><span class="dv">1000</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw">length</span>(all_studies)</a></code></pre></div>
<pre><code>## [1] 382</code></pre>
<p>To get a list of traits, use <code>list_traits()</code>. Again the default is to return just 20 values. To get all traits, use the <code>size</code> argument.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">all_traits &lt;-<span class="st"> </span><span class="kw">list_traits</span>(<span class="dt">size=</span><span class="dv">1000</span>)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw">length</span>(all_traits)</a></code></pre></div>
<pre><code>## [1] 382</code></pre>
<p>The traits are returned as identifiers like EFO_0001360. To get a description of a trait, you can use the ontology lookup service, for example <a href="https://www.ebi.ac.uk/efo/EFO_0001360" class="uri">https://www.ebi.ac.uk/efo/EFO_0001360</a></p>
<p>Note that the traits returned are not all distinct.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw">table</span>(<span class="kw">table</span>(all_traits))</a></code></pre></div>
<pre><code>## 
##   1   2   3   4   5   6   8   9  10 
## 177  30  18   5   4   4   1   1   1</code></pre>
<p>Chromosomes are stored as integers 1-24.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">list_chr</span>()</a></code></pre></div>
<pre><code>##  [1]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24</code></pre>
</div>
<div id="get-associations" class="section level2">
<h2>Get associations</h2>
<p>To get associations for a specific variant by its rs-number, use <code>get_variant()</code>. If you know the chromosome it is on, youâ€™ll get faster results by providing the chromosome. And again, the default is to return just 20 values, so use the <code>size</code> and <code>start</code> arguments if you want a comprehensive list.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">result &lt;-<span class="st"> </span><span class="kw">get_variant</span>(<span class="st">&quot;rs2228603&quot;</span>, <span class="dv">19</span>, <span class="dt">size=</span><span class="dv">5</span>)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">result[,<span class="kw">c</span>(<span class="st">&quot;p_value&quot;</span>, <span class="st">&quot;study_accession&quot;</span>, <span class="st">&quot;trait&quot;</span>)]</a></code></pre></div>
<pre><code>##      p_value study_accession       trait
## 0 0.02980604      GCST000392 EFO_0001359
## 1 0.39340000      GCST000568 EFO_0004465
## 2 0.39340000      GCST000568 EFO_0004465
## 3 0.60860000      GCST000569 EFO_0004307
## 4 0.29170000      GCST000571 EFO_0004466</code></pre>
<p>Use the arguments <code>p_lower</code> and <code>p_upper</code> to focus on associations with p-value in a specified range. For example, to get all of the associations with p-value &lt; 10<sup>-10</sup>, you would do:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">result &lt;-<span class="st"> </span><span class="kw">get_variant</span>(<span class="st">&quot;rs2228603&quot;</span>, <span class="dv">19</span>, <span class="dt">p_upper=</span><span class="fl">1e-10</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">result[,<span class="kw">c</span>(<span class="st">&quot;p_value&quot;</span>, <span class="st">&quot;study_accession&quot;</span>, <span class="st">&quot;trait&quot;</span>)]</a></code></pre></div>
<pre><code>##     p_value study_accession       trait
## 0 1.739e-57      GCST002216 EFO_0004530
## 1 1.739e-57      GCST002216 EFO_0004530
## 2 1.049e-62      GCST002221 EFO_0004574
## 3 1.049e-62      GCST002221 EFO_0004574
## 4 4.433e-44      GCST002222 EFO_0004611
## 5 4.433e-44      GCST002222 EFO_0004611</code></pre>
<p>To get associations for a specific region, use <code>get_asso()</code>. For example, to get the region from 19.2 Mbp to 19.3 Mbp on chr 19:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">result &lt;-<span class="st"> </span><span class="kw">get_asso</span>(<span class="dt">chr=</span><span class="dv">19</span>, <span class="dt">bp_lower=</span><span class="dv">19200000</span>, <span class="dt">bp_upper=</span><span class="dv">19300000</span>)</a>
<a class="sourceLine" id="cb18-2" data-line-number="2">result[,<span class="kw">c</span>(<span class="st">&quot;chromosome&quot;</span>, <span class="st">&quot;base_pair_location&quot;</span>, <span class="st">&quot;p_value&quot;</span>, <span class="st">&quot;study_accession&quot;</span>, <span class="st">&quot;trait&quot;</span>)]</a></code></pre></div>
<pre><code>##    chromosome base_pair_location    p_value study_accession       trait
## 0          19           19208675 0.34851112      GCST000392 EFO_0001359
## 1          19           19219115 0.02980604      GCST000392 EFO_0001359
## 2          19           19224274 0.72647403      GCST000392 EFO_0001359
## 3          19           19225799 0.09019226      GCST000392 EFO_0001359
## 4          19           19230637 0.59469796      GCST000392 EFO_0001359
## 5          19           19233770 0.93955453      GCST000392 EFO_0001359
## 6          19           19247863 0.76167551      GCST000392 EFO_0001359
## 7          19           19250926 0.24164792      GCST000392 EFO_0001359
## 8          19           19274602 0.71375546      GCST000392 EFO_0001359
## 9          19           19277637 0.43687139      GCST000392 EFO_0001359
## 10         19           19280236 0.76699603      GCST000392 EFO_0001359
## 11         19           19281592 0.70582591      GCST000392 EFO_0001359
## 12         19           19297355 0.78331140      GCST000392 EFO_0001359
## 13         19           19207349 0.83550000      GCST000568 EFO_0004465
## 14         19           19207349 0.83550000      GCST000568 EFO_0004465
## 15         19           19208675 0.91250000      GCST000568 EFO_0004465
## 16         19           19208675 0.91250000      GCST000568 EFO_0004465
## 17         19           19213634 0.73850000      GCST000568 EFO_0004465
## 18         19           19213634 0.73850000      GCST000568 EFO_0004465
## 19         19           19219115 0.39340000      GCST000568 EFO_0004465</code></pre>
<p>You can restrict those results to a particular study.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">result &lt;-<span class="st"> </span><span class="kw">get_asso</span>(<span class="dt">chr=</span><span class="dv">19</span>, <span class="dt">bp_lower=</span><span class="dv">19200000</span>, <span class="dt">bp_upper=</span><span class="dv">19300000</span>, <span class="dt">study=</span><span class="st">&quot;GCST000392&quot;</span>)</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">result[,<span class="kw">c</span>(<span class="st">&quot;chromosome&quot;</span>, <span class="st">&quot;base_pair_location&quot;</span>, <span class="st">&quot;p_value&quot;</span>, <span class="st">&quot;study_accession&quot;</span>, <span class="st">&quot;trait&quot;</span>)]</a></code></pre></div>
<pre><code>##    chromosome base_pair_location    p_value study_accession       trait
## 0          19           19208675 0.34851112      GCST000392 EFO_0001359
## 1          19           19219115 0.02980604      GCST000392 EFO_0001359
## 2          19           19224274 0.72647403      GCST000392 EFO_0001359
## 3          19           19225799 0.09019226      GCST000392 EFO_0001359
## 4          19           19230637 0.59469796      GCST000392 EFO_0001359
## 5          19           19233770 0.93955453      GCST000392 EFO_0001359
## 6          19           19247863 0.76167551      GCST000392 EFO_0001359
## 7          19           19250926 0.24164792      GCST000392 EFO_0001359
## 8          19           19274602 0.71375546      GCST000392 EFO_0001359
## 9          19           19277637 0.43687139      GCST000392 EFO_0001359
## 10         19           19280236 0.76699603      GCST000392 EFO_0001359
## 11         19           19281592 0.70582591      GCST000392 EFO_0001359
## 12         19           19297355 0.78331140      GCST000392 EFO_0001359</code></pre>
<p>To get associations for a given trait, use <code>get_trait_asso()</code>. You canâ€™t restrict this to a given chromosome region.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">result &lt;-<span class="st"> </span><span class="kw">get_trait_asso</span>(<span class="st">&quot;EFO_0001360&quot;</span>, <span class="dt">p_upper=</span><span class="fl">1e-100</span>, <span class="dt">size=</span><span class="dv">1000</span>)</a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="kw">nrow</span>(result)</a></code></pre></div>
<pre><code>## [1] 106</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">result[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="kw">c</span>(<span class="st">&quot;chromosome&quot;</span>, <span class="st">&quot;base_pair_location&quot;</span>, <span class="st">&quot;p_value&quot;</span>, <span class="st">&quot;study_accession&quot;</span>, <span class="st">&quot;trait&quot;</span>)]</a></code></pre></div>
<pre><code>##   chromosome base_pair_location   p_value study_accession       trait
## 0         10          112988738 4.38e-109      GCST006867 EFO_0001360
## 1         10          112988858 4.96e-127      GCST006867 EFO_0001360
## 2         10          112989975 2.98e-221      GCST006867 EFO_0001360
## 3         10          112990477 7.82e-202      GCST006867 EFO_0001360
## 4         10          112990621 2.17e-202      GCST006867 EFO_0001360</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
